requestparsing/test_harness.go

```diff
@@ -0,0 +1,113 @@
+package requestparsing
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net"
+	"net/http"
+	"sync"
+)
+
+type mockHandler struct {
+	// called in the ServeHTTP function with the received request.
+	callback func(*http.Request)
+}
+
+func (h *mockHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	h.callback(r)
+	io.WriteString(w, "Hello world!")
+}
+
+type mockListener struct {
+	closeOnce      sync.Once
+	connChannel    chan net.Conn
+	serverEndpoint io.Closer
+	clientEndpoint net.Conn
+}
+
+// Creates a mock listener that passes requests to the HTTP server as part of
```

Comment:
Function docstrings should begin with the name of the function and end with punctuation.

// newMockListener creates a ... as part of the test harness.

Also: are these fakes instead of mocks?

Reply:
Docs: https://github.com/golang/go/wiki/CodeReviewComments#doc-comments

Reply:
I think they are fakes, renamed in [beb7c84](url) and refactored documentation.